rosidl_default_runtime/QUALITY_DECLARATION.md

```diff
@@ -86,7 +86,7 @@ Results of linter tests can be found [here](https://ci.ros2.org/view/nightly/job
 
 ### Direct Runtime ROS Dependencies [5.i]/[5.ii]
 
-`rosidl_default_runtime` has the following group dependencies:
+`rosidl_default_runtime` has the following group dependencies all of which are at **Quality Level 1** for **Quality Level 1** middlewares:
 * `rosidl_runtime_packages`
 * `rosidl_typesupport_c_packages`
 * `rosidl_typesupport_cpp_packages`
```

Comment:
I don't have a lot of experience with group dependencies, so I just wanted to double-check that there isn't more information that we could provide to the reader here. How can they verify this claim for a given RMW?

Reply:
Whatever solution we come up with, it will also apply to all the packages in `rcl_interfaces` and `common_interfaces`. Colcon doesn't have a way to list dependencies by group as far as I can tell. You can grep package.xmls in the src directory for the `<member_of_group>rosidl_default_runtime`. 

Since the group could potentially change, we could say something like, "This group as of writing currently contains the following packages, but that may change in the future"

Reply:
I just did some searches in the `ros2` org:

## [rosidl_runtime_packages](https://github.com/search?q=org%3Aros2+member_of_group+rosidl_runtime_packages&type=code)

This group doesn't seem to be vendor-dependent. Do you know if this is generally the case? If so, I think we could list the packages here, and update the list whenever more are added?

These are the packages:

* `rosidl_typesupport_c` (QL1)
* `rosidl_typesupport_cpp` (QL1)
* `rosidl_runtime_c` (QL1)
* `rosidl_runtime_cpp` (QL1)

## [rosidl_typesupport_c_packages](https://github.com/search?q=org%3Aros2+member_of_group+rosidl_typesupport_c_packages&type=code)

That seems to include a `rosidl_typesupport_X_c` for a given X RMW. So maybe we could instruct readers to look for a package like this for their RMW?

## [rosidl_typesupport_cpp_packages](https://github.com/search?q=org%3Aros2+member_of_group+rosidl_typesupport_cpp_packages&type=code)

Same as above, that seems to include a `rosidl_typesupport_X_cpp` for a given X RMW.

I also see `rosidl_typesupport_introspection_cpp`, which doesn't have a quality declaration. Do we need to justify why that package is not required for quality purposes?

Reply:
Ok, I took a stab at updating this section. Let me know what you think.

Reply:
Looking good! Just checking, the bit about Python is the one that covers `rosidl_typesupport_introspection_cpp`?

Reply:
I believe that sentence applies to rosidl_default_runtime as a whole.

Reply:
Sorry I'm not being clear.

`rosidl_typesupport_introspection_cpp` [is part of](https://github.com/ros2/rosidl/blob/1458d50450823ceeae0399d17c7e5c6c56c1928f/rosidl_typesupport_introspection_cpp/package.xml#L33) the `rosidl_typesupport_cpp_packages` group, but it's not middleware-specific. The current QD reads:

```
* `rosidl_typesupport_cpp_packages` At the time of writing, this group does not contain middleware independent runtime dependencies. 
```

Does that mean that `rosidl_typesupport_introspection_cpp` isn't a runtime dependency?

Reply:
I don't think so. It is a generator package that generates code from python empy templates for each interface package. But the generated code is all static object definitions. So it's all build time dependencies.

Reply:
Got it, thanks, this is the clarification I was looking for :+1: